packages/eslint-plugin/src/test-utils.ts

```diff
@@ -0,0 +1,70 @@
+import { basename } from 'path';
+
+import { RuleTester } from 'eslint';
+
+(RuleTester as any).describe = (text: string, method: (...args: any[]) => void) => {
+  const origHasAssertions = expect.hasAssertions;
+  describe(text, () => {
+    beforeAll(() => {
+      // Stub out expect.hasAssertions beforeEach from jest-presetup.js
+      expect.hasAssertions = () => {};
+    });
+    afterAll(() => {
+      expect.hasAssertions = origHasAssertions;
+    });
+
+    method();
+  });
+};
+
+export const tester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+export const createAliasedInvalidTestCase = (
+  test: RuleTester.InvalidTestCase,
+  replaceCode: (code: string) => string,
+  replaceOutput: (output: string) => string
+): RuleTester.InvalidTestCase => ({
+  ...test,
+  filename: `aliased-${basename(test.filename!)}.ts`,
+  code: replaceCode(test.code),
+  output: replaceOutput(test.output!),
+});
+
+export const createDeclarationInvalidTestCases = (
+  test: RuleTester.InvalidTestCase,
+  name: string,
+  replaceCode: (code: string, prefix: string) => string,
+  replaceOutput: (output: string, prefix: string) => string
+): RuleTester.InvalidTestCase[] => {
+  const filename = basename(test.filename!);
+  const exportDefaultPrefix = 'export default ';
+  const exportNamedPrefix = `export const ${name} = `;
```

Comment:
Do we have to assume there is trailing whitespace after the `=`?  If consumer doesn't use a code formatter or right side is on a new line will replacement still work?

Reply:
Yes, it will still work because the rules detect the tagged template expression directly. The declaration variants here simply validate that the indentation is correct, and the object properties start just after `export`, `const`, or the tagged template.